File: nvim/lua/plugins.lua
local execute = vim.api.nvim_command
local fn = vim.fn

local install_path = fn.stdpath('data')..'/site/pack/packer/opt/packer.nvim'

if fn.empty(fn.glob(install_path)) > 0 then
  execute('!git clone https://github.com/wbthomason/packer.nvim '..install_path)
  execute 'packadd packer.nvim'
end

vim.cmd [[packadd packer.nvim]]

return require('packer').startup(function()
	use {'wbthomason/packer.nvim', opt = true}

	use { 'aklt/plantuml-syntax' }

	use {'kyazdani42/nvim-web-devicons'}

	use { 'nvim-treesitter/nvim-treesitter', run = ':TSUpdate' }
	use { 'neovim/nvim-lspconfig' }
	use { 'nvim-lua/lsp-status.nvim' }

	use { 'kosayoda/nvim-lightbulb' }
	use { 'Yggdroot/indentLine'} 
	use {
  		'lewis6991/gitsigns.nvim',
  		requires = {
    			'nvim-lua/plenary.nvim'
  		},
		config = function()
    			require('gitsigns').setup()
  		end
	}

	use { 'bluz71/vim-moonfly-colors' }
	use { 'hrsh7th/vim-vsnip' }
	use { 'hrsh7th/nvim-compe' }

	use { 'dpelle/vim-Grammalecte' }
	use { 'dpelle/vim-LanguageTool' }

	use {'nvim-lua/lsp_extensions.nvim'}
	use {'ray-x/lsp_signature.nvim'}

	use {
  		'nvim-telescope/telescope.nvim',
  		requires = {{'nvim-lua/popup.nvim'}, {'nvim-lua/plenary.nvim'}}
	}

	use 'glepnir/galaxyline.nvim'
	use 'drmikehenry/vim-headerguard'
	use 'andymass/vim-matchup'
	use 'b3nj5m1n/kommentary'
	use 'onsails/lspkind-nvim'
end)

File: plugins.py
plugins = [
    "aklt/plantuml-syntax",
    "kyazdani42/nvim-web-devicons",
    "nvim-treesitter/nvim-treesitter",
    "neovim/nvim-lspconfig",
    "nvim-lua/lsp-status.nvim",
    "kosayoda/nvim-lightbulb",
    "Yggdroot/indentLine",
    "lewis6991/gitsigns.nvim",
    "nvim-lua/plenary.nvim",
    "bluz71/vim-moonfly-colors",
    "hrsh7th/vim-vsnip",
    "hrsh7th/cmp-nvim-lsp",
    "hrsh7th/cmp-buffer",
    "hrsh7th/cmp-calc",
    "hrsh7th/cmp-path",
    "kdheepak/cmp-latex-symbols",
    "hrsh7th/nvim-cmp",
    "dpelle/vim-Grammalecte",
    "dpelle/vim-LanguageTool",
    "nvim-lua/lsp_extensions.nvim",
    "ray-x/lsp_signature.nvim",
    "nvim-telescope/telescope.nvim",
    "nvim-lua/popup.nvim",
    "NTBBloodbath/galaxyline.nvim",
    "drmikehenry/vim-headerguard",
    "andymass/vim-matchup",
    "b3nj5m1n/kommentary",
    "onsails/lspkind-nvim",
    "editorconfig/editorconfig-vim",
    "jose-elias-alvarez/null-ls.nvim",
    "nathom/filetype.nvim",
    "tweekmonster/startuptime.vim",
]

inputs = ""
nvim_plugins = "nvim-plugins = [\n";
for plugin in plugins:
    repo,plugin = plugin.split('/')
    inputs += f'"{plugin}" = {{\n'
    inputs += f'  url = "github:{repo}/{plugin}";\n'
    inputs += f'  flake = false;\n'
    inputs += "};\n"

    nvim_plugins += "  {\n"
    nvim_plugins += f'    name = "{plugin}";\n'
    nvim_plugins += f'    path = inputs."{plugin}";\n'
    nvim_plugins += "  }\n"
nvim_plugins += "];\n"

with open("flake.nix.in") as f:
    flake = f.read()
    flake = flake.replace("@inputs@\n", inputs)
    flake = flake.replace("@plugins@\n", nvim_plugins)
    out = open("flake.nix", "w")
    out.write(flake)

